<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BalanceRobot</class>
 <widget class="QMainWindow" name="BalanceRobot">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1174</width>
    <height>554</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BalanceRobot</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>0</y>
      <width>521</width>
      <height>491</height>
     </rect>
    </property>
    <property name="title">
     <string>PID Graph</string>
    </property>
    <widget class="QGraphicsView" name="pidGraph">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>501</width>
       <height>401</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="btnExport">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>460</y>
       <width>51</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Export</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="filePathEdit">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>460</y>
       <width>191</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>c:/aaa.txt</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnSelectFile">
     <property name="geometry">
      <rect>
       <x>374</x>
       <y>460</y>
       <width>81</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Select File</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnPlotP">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>430</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Times New Roman</family>
       <pointsize>12</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: green</string>
     </property>
     <property name="text">
      <string>PTerm</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnPlotI">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>430</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Times New Roman</family>
       <pointsize>12</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(170, 170, 0)</string>
     </property>
     <property name="text">
      <string>ITerm</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnPlotD">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>430</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Times New Roman</family>
       <pointsize>12</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: magenta</string>
     </property>
     <property name="text">
      <string>DTerm</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnPlotV">
     <property name="geometry">
      <rect>
       <x>410</x>
       <y>430</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Times New Roman</family>
       <pointsize>12</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: rgb(255, 170, 0)</string>
     </property>
     <property name="text">
      <string>STerm</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnPlotOutput">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>430</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Times New Roman</family>
       <pointsize>12</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: blue</string>
     </property>
     <property name="text">
      <string>Output</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnPlotInput">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>430</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Times New Roman</family>
       <pointsize>12</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">color: red</string>
     </property>
     <property name="text">
      <string>Input</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnClearBuf">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>460</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear Buf</string>
     </property>
    </widget>
    <widget class="QLabel" name="txt_bufcnt">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>460</y>
       <width>54</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>0</y>
      <width>291</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Communication</string>
    </property>
    <widget class="QComboBox" name="serialBox">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>20</y>
       <width>161</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>20</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Serial:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>13</x>
       <y>50</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Baud:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="baudBox">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>50</y>
       <width>81</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="refreshButton">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>50</y>
       <width>71</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Refresh</string>
     </property>
    </widget>
    <widget class="QPushButton" name="connectButton">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>80</y>
       <width>161</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Connect</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="arm0_groupBox">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>110</y>
      <width>291</width>
      <height>381</height>
     </rect>
    </property>
    <property name="title">
     <string>Parameters</string>
    </property>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>250</y>
       <width>151</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Sample Time(ms):</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="sample_time">
     <property name="geometry">
      <rect>
       <x>176</x>
       <y>250</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>270</y>
       <width>151</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Falling Angle:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="falling_angle">
     <property name="geometry">
      <rect>
       <x>176</x>
       <y>270</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>290</y>
       <width>151</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Balance Angle(setpoint):</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="balance_angle">
     <property name="geometry">
      <rect>
       <x>176</x>
       <y>290</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="btnRead">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>350</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Read</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnApply">
     <property name="geometry">
      <rect>
       <x>96</x>
       <y>350</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Apply</string>
     </property>
    </widget>
    <widget class="QPushButton" name="btnWrite">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>350</y>
       <width>61</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Write</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkEnable">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>320</y>
       <width>121</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Enable Balancing</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>251</width>
       <height>41</height>
      </rect>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">border: 2px solid gray; </string>
     </property>
     <property name="title">
      <string/>
     </property>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>P</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="param_i">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>10</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>10</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>D</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>10</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>I</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="param_d">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>10</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="param_p">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>10</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>111</width>
       <height>61</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border: 2px solid gray; </string>
     </property>
     <property name="title">
      <string/>
     </property>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>21</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Ep</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="param_posScaler">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>10</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="param_speedScaler">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>30</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>21</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Es</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>70</y>
       <width>151</width>
       <height>61</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">border: 2px solid gray; </string>
     </property>
     <property name="title">
      <string/>
     </property>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Output Min</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="param_max">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>10</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="param_min">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>30</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Output Max</string>
      </property>
     </widget>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>170</y>
       <width>151</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Left Deadband:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="deadL">
     <property name="geometry">
      <rect>
       <x>176</x>
       <y>170</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_18">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>190</y>
       <width>151</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Right Deadband:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="deadR">
     <property name="geometry">
      <rect>
       <x>176</x>
       <y>190</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QCheckBox" name="pidCheckBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>150</y>
       <width>161</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Calc ZieglerNicolas PID</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="accele_time">
     <property name="geometry">
      <rect>
       <x>176</x>
       <y>210</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_19">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>210</y>
       <width>151</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Accelerate Time(x10ms):</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="accele_angle">
     <property name="geometry">
      <rect>
       <x>176</x>
       <y>230</y>
       <width>51</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_20">
     <property name="geometry">
      <rect>
       <x>16</x>
       <y>230</y>
       <width>151</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Accelerate Angle:</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_6">
    <property name="geometry">
     <rect>
      <x>840</x>
      <y>0</y>
      <width>321</width>
      <height>331</height>
     </rect>
    </property>
    <property name="title">
     <string>Tripple Balance Vector</string>
    </property>
    <widget class="QGraphicsView" name="vectroGraph">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>301</width>
       <height>301</height>
      </rect>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1174</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
